src/utility.py

```diff
@@ -0,0 +1,18 @@
+import secrets
+
+
+def gaurd_edge(function):
```

Comment:
IMHO it's name is not appropriate and it might not be necessary at this state, we may get back to it  later soon.